### PR TITLE
chore: Cleanup full raw avatar responses from tRPC endpoints

### LINKED ISSUE #8044: [CAL-1383] Cleanup full raw avatar responses from tRPC endpoints
Found one in `packages/trpc/server/routers/viewer/eventTypes.ts`

This response:

Should be:

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1383](https://linear.app/calcom/issue/CAL-1383/cleanup-full-raw-avatar-responses-from-trpc-endpoints)</sub>

### PR DESCRIPTION
**Pull request by asadath1395**
## What does this PR do?

[CAL-1383] Cleanup full raw avatar responses from tRPC endpoints

Fixes #8044 

**Environment**: Staging(main branch) / Production

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)

### CODE REVIEW

**Review by alannnc — APPROVED** [review]

File: packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts [hunk 1]
```diff
@@ -172,13 +172,19 @@ export const getByViewerHandler = async ({ ctx }: GetByViewerOptions) => {
   const mergedEventTypes = Object.values(eventTypesHashMap)
     .map((eventType) => eventType)
     .filter((evType) => evType.schedulingType !== SchedulingType.MANAGED);
+
+  let image = user.avatar || undefined
+  if (image && user?.username) {
+    image = `${CAL_URL}/${user.username}/avatar.png`
+  }
```
Comment (roae) [inline packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts:179]: We can simplify this
```suggestion
  const image = user?.username ? `${CAL_URL}/${user.username}/avatar.png` : undefined;
```

Comment (asadath1395) [reply packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts:179]: @roae Its not guaranteed users will always have avatar, no? Again we could add more condition but code will become messy

**Comment by PeerRich** [comment]
i think some tests failing, not sure. lets see if it passes now

File: packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts (hunk 1 above)
Comment (roae) [reply packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts:179]: @asadath1395 The avatar.png endpoint show a placeholder if the user hasn’t avatar

**Comment by PeerRich** [comment]
hey @asadath1395 there are some lint issues https://github.com/calcom/cal.com/actions/runs/5090502282/jobs/9149400853?pr=8901

**Comment by asadath1395** [comment]
@roae I have addressed your comment. Please re-review. Thanks

**Review by emrysal — APPROVED** [review]

**Review by keithwillcode — CHANGES_REQUESTED** [review]
There are type checks failing

**Review by keithwillcode — APPROVED** [review]

**Pull request merged**